lib/host/wasi/wasifunc.cpp

```diff
@@ -1525,6 +1524,164 @@ Expect<uint32_t> WasiSchedYield::body(Runtime::Instance::MemoryInstance *) {
   return __WASI_ERRNO_SUCCESS;
 }
 
+Expect<uint32_t> WasiSockOpen::body(Runtime::Instance::MemoryInstance *MemInst,
+                                    int32_t AddressFamily, int32_t SockType,
+                                    uint32_t RoFdPtr) {
+  /// Check memory instance from module.
+  if (MemInst == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+
+  __wasi_fd_t *const RoFd =
+      MemInst->getPointer<__wasi_fd_t *>(RoFdPtr, sizeof(__wasi_fd_t));
+  if (RoFd == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+  const __wasi_address_family_t family =
+      static_cast<__wasi_address_family_t>(AddressFamily);
+  const __wasi_sock_type_t sock_type =
+      static_cast<__wasi_sock_type_t>(SockType);
+  if (auto Res = Env.sockOpen(family, sock_type); unlikely(!Res)) {
+    return Res.error();
+  } else {
+    *RoFd = *Res;
+  }
+
+  return __WASI_ERRNO_SUCCESS;
+}
+
+Expect<uint32_t> WasiSockClose::body(Runtime::Instance::MemoryInstance *MemInst,
+                                     int32_t Fd) {
+  /// Check memory instance from module.
+  if (MemInst == nullptr || Fd < 0) {
```

Comment:
1. No need to check `MemInst`
2. `Fd` should be checked at `Env.sockClose`
See include/host/wasi/environ.h, `fdClose` function

Reply:
Sorry, this interface is not used. We should use sockshutdown and fdclose, and this api should be deleted.